webapp/application.py

```diff
@@ -39,7 +39,9 @@
 def _get_application(application_id):
     application = harvest.get_application(int(application_id))
     job_post_id = application["job_post_id"]
-    application["job_post"] = harvest.get_job_post(int(job_post_id))
+    application["job_post"] = (
+        harvest.get_job_post(int(job_post_id)) if job_post_id else None
```

Comment:
Another way of writing this is:
```py
if job_post_id:
  application["field"] = value
```
and casting to `int()` is not required.

But it's up to you, which style you prefer.

Reply:
Removed the `int`, thanks!

I prefer your style. The problem is that you don't set the value explicitly to `None`, `job_post` won't exist in the application dict, and trying to access it will result in a `KeyError`. Or I think so...